geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegionMap.java

```diff
@@ -93,7 +93,7 @@
    * This test hook is used to force the conditions for defect 48182. This hook is used by
    * Bug48182JUnitTest.
    */
-  static Runnable testHookRunnableFor48182 = null;
+  static volatile Runnable testHookRunnableFor48182 = null;
```

Comment:
Test hooks in production code are pretty smelly.  Can this be refactored to use a Mockito mock in the associated test instead and remove this hook outright?

Reply:
I agree, test hooks are bad...We thought about replacing this with mock, powerMock, but with our understanding of the problem, it may be harder to mock it and reproduce the issue. For now, we have made changes in the existing test code to root cause the issue. 
Also, the testhooks are broader issue. I have created a ticket to address this in the product.
https://issues.apache.org/jira/browse/GEODE-3681